File: algorithms/1_easy/G/01374_generate_a_string_with_characters_that_have_odd_counts/solution.cpp
class Solution {
public:
    string generateTheString(int n) {
        string res(static_cast<size_t>(n), 'a');

        if (n % 2 == 0) {
            *(res.end() - 1) = 'b';
        }

        return res;
    }
};

File: algorithms/1_easy/G/01374_generate_a_string_with_characters_that_have_odd_counts/solution.rs
impl Solution {
    pub fn generate_the_string(n: i32) -> String {
        let mut vec = vec![b'a'; n as usize];

        if n % 2 == 0 {
            *vec.last_mut().unwrap() = b'b';
        }

        String::from_utf8(vec).unwrap()
    }
}
